.github/workflows/cpp.yml

```diff
@@ -175,10 +175,15 @@ jobs:
       ARROW_BUILD_STATIC: OFF
       ARROW_BOOST_USE_SHARED: OFF
       ARROW_VERBOSE_THIRDPARTY_BUILD: OFF
-      BOOST_SOURCE: BUNDLED
+      BOOST_ROOT: C:\local\boost_1_67_0
+      BOOST_LIBRARYDIR: C:\local\boost_1_67_0\lib64-msvc-14.1\
```

Comment:
Should we update this when `boost-msvc-14.1` is updated?
Can we detect this automatically?

Does this work?

```diff
diff --git a/.github/workflows/cpp.yml b/.github/workflows/cpp.yml
index abc3b8f4d..4cccbdfd0 100644
--- a/.github/workflows/cpp.yml
+++ b/.github/workflows/cpp.yml
@@ -185,7 +185,11 @@ jobs:
           submodules: true
       - name: Build
         shell: bash
-        run: ci/scripts/cpp_build.sh $(pwd) $(pwd)/build
+        run: |
+          boost_root="$(echo /c/local/boost_* | head -n1)"
+          export BOOST_ROOT="c:\\local\\$(dirname ${boost_root})"
+          export BOOST_LIBRARYDIR="${BOOST_ROOT}\\lib64-msvc-14.1"
+          ci/scripts/cpp_build.sh $(pwd) $(pwd)/build
       - name: Test
         shell: bash
         run: ci/scripts/cpp_test.sh $(pwd) $(pwd)/build
```

Reply:
Hmm, I'll try it :-)

Reply:
I did this:
```
  export BOOST_ROOT="$(ls -1 /c/local/boost_* | tail -n1)"
  export BOOST_LIBRARYDIR="${BOOST_ROOT}\\lib64-msvc-14.1"
  echo "BOOST_ROOT: ${BOOST_ROOT}"
```
and I got:
```
BOOST_ROOT: tools
```
:-O

I don't want to spend time debugging this (especially as Github Actions seems a bit slow today), so I'm gonna revert the change.

Reply:
OK. I'll take it over as a follow-up task.